HLTrigger/JetMET/plugins/HLTJetHFCleaner.h

```diff
@@ -0,0 +1,221 @@
+      dphiJetMetMin_(iConfig.getParameter<double>("dphiJetMetMin")),
+      jetEtaMin_(iConfig.getParameter<double>("jetEtaMin")),
+      jetEtaMax_(iConfig.getParameter<double>("jetEtaMax")),
+      sigmaEtaPhiDiffMax_(iConfig.getParameter<double>("sigmaEtaPhiDiffMax")),
+      centralEtaStripSizeMax_(iConfig.getParameter<int>("centralEtaStripSizeMax")),
+      cornerCutSigmaEtaEta_(iConfig.getParameter<double>("cornerCutSigmaEtaEta")),
+      cornerCutSigmaPhiPhi_(iConfig.getParameter<double>("cornerCutSigmaPhiPhi")),
+      applySigmaEtaPhiCornerCut_(iConfig.getParameter<bool>("applySigmaEtaPhiCornerCut")),
+      applySigmaEtaPhiCut_(iConfig.getParameter<bool>("applySigmaEtaPhiCut")),
+      applyStripSizeCut_(iConfig.getParameter<bool>("applyStripSizeCut")) {
+  produces<JetCollection>();
+}
+
+template <typename JetType>
+void HLTJetHFCleaner<JetType>::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup&) const {
+  auto const jets = iEvent.getHandle(jetToken_);
+  auto const& mets = iEvent.get(metToken_);
+
+  // The set of cleaned jets
+  auto cleaned_jets = std::make_unique<JetCollection>();
+  cleaned_jets->reserve(jets->size());
+
+  // Check if length of the MET vector is 1, throw exception otherwise
+  if (mets.size() != 1) {
+    throw cms::Exception("ComparisonFailure") << "Size of reco::MET collection is different from 1";
+  }
+  auto const met_phi = mets.front().phi();
+
+  auto const& sigmaEtaEtas = iEvent.get(sigmaEtaEtaToken_);
+  auto const& sigmaPhiPhis = iEvent.get(sigmaPhiPhiToken_);
+  auto const& centralEtaStripSizes = iEvent.get(centralEtaStripSizeToken_);
+
+  // Loop over the jets and do cleaning
+  for (uint ijet = 0; ijet < jets->size(); ijet++) {
+    auto const jet = (*jets)[ijet];
+    auto const withinEtaRange = (std::abs(jet.eta()) < jetEtaMax_) and (std::abs(jet.eta()) > jetEtaMin_);
+    // Cuts are applied to jets that are back to back with MET
+    if (std::abs(reco::deltaPhi(jet.phi(), met_phi)) <= dphiJetMetMin_) {
+      cleaned_jets->emplace_back(jet);
+      continue;
+    }
+    // If the jet is outside the range that we're interested in,
+    // append it to the cleaned jets collection and continue
+    if ((jet.pt() < jetPtMin_) or (!withinEtaRange)) {
+      cleaned_jets->emplace_back(jet);
+      continue;
+    }
+
+    JetRef const jetRef(jets, ijet);
+
+    // Cuts related to eta and phi width of HF showers:
+    // Sigma eta-phi based cut + central strip size cut
+    // We'll apply the sigma eta+phi dependent cut only if requested via the config file
+    if (applySigmaEtaPhiCut_) {
+      // Sigma eta eta and sigma phi phi
+      auto const sigmaEtaEta = sigmaEtaEtas[jetRef];
+      auto const sigmaPhiPhi = sigmaPhiPhis[jetRef];
+
+      // Check if sigma eta eta and sigma phi phi are both set to -1.
+      // If this is the case, we can skip the jet without evaluating any mask.
+      if ((sigmaEtaEta == -1) and (sigmaPhiPhi == -1)) {
```

Comment:
I don't know if this can happen here, but in principle this comparison between floats could fail if the valueMap returns something like `-1.0000002` (for floating-point precision, or whatever other reason). Would it make sense to ask for `< 0` instead?

Reply:
Ah makes much more sense, sure I'll do that, thanks!

Reply:
I did this, squashed that last commit to the previous one, and pushed the merged commit, [1535d04](https://github.com/cms-sw/cmssw/pull/37539/commits/1535d04aa40544ecdf0579075ff871e61f4b718f). Let me know if you have more comments, thanks!